### PR TITLE
[pylint] Checker for exception logging pylint

### LINKED ISSUE #3227: [Pylint] Custom Linter check for Exception Logging
The SDKs should not be logging `Exception` and rely on the policy set in Azure Core (only when level is set to `DEBUG` and `is_enabled` is `True`).  We want to look for this to avoid inadvertently logging PII information that might be in a message/body of an unexpected exception. 

We want to look for `except Exception as ex` scenarios and if the exception is being logged right after and raise an error.

### PR DESCRIPTION
**Pull request by l0lawrence**
Would like to test this on sdks/through API View before it becomes a definitive checker in CI

Fixes #3227

### CODE REVIEW

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 1]
```diff
@@ -1916,6 +1916,62 @@ def visit_module(self, node):
...
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(CheckExceptionLogging, self).__init__(linter)
+
+    def visit_tryexcept(self, node):
+        logger_names = ["_logger", "logger"]
+        uses_exception = False
+        if isinstance(node.handlers[0], astroid.ExceptHandler) and node.handlers[0].type:
```
Comment (catalinaperalta) [inline tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:1950]: Why does this have to check that the handler has a type on it? Are there cases where it doesnt?

Comment (catalinaperalta) [reply tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:1950]: Also is it always the first handler?

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 2]
```diff
@@ -1916,6 +1916,62 @@ def visit_module(self, node):
...
+                        uses_exception=True
+            else:
+                if node.handlers[0].type.name=="Exception":
+                    uses_exception=True
+                    
+            handler_name = "Exception"
+            if uses_exception: 
+                # If they have renamed the exception
+                if node.handlers[0].name:
+                    handler_name = node.handlers[0].name.name
```
Comment (catalinaperalta) [inline tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:1964]: Why does this have `.name` twice?

**Review by catalinaperalta — COMMENTED** [review]
Did you run this on SDK packages?

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py (hunk 2 above)
Comment (l0lawrence) [reply tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:1964]: name is an object called AssignName that has another name param that is a string

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py (hunk 1 above)
Comment (l0lawrence) [reply tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:1950]: I will investigate!

**Pull request closed without merging**